### ISSUE TITLE
Extra indent for soft-wrapped lines

### ISSUE THREAD

**Issue by andreyvit** [zed-industries/zed#18687]
### Check for existing issues

- [X] Completed

### Describe the feature

A soft-wrapped line is displayed with the same indentation as its beginning, which is hard to read. Other editors have options to add 1 or 2 indent sizes to these lines.

Proposal: add `extra_soft_wrap_indent` option.

Compare Zed:

<img width="587" alt="Screenshot 2024-10-03 at 1 28 58 PM" src="https://github.com/user-attachments/assets/40dc4061-cc3e-4c02-9759-ea6768f4b74e">

and say Sublime Text:

<img width="585" alt="Screenshot 2024-10-03 at 1 29 12 PM" src="https://github.com/user-attachments/assets/e55c69c4-5fa8-4d95-922e-d3e559ce0226">

**Comment by JosephTLyons** [comment]
Actually, I just saw another issue you made here:

- https://github.com/zed-industries/zed/issues/18686

Assuming all is the same:

Zed: v0.155.2 (Zed)
OS: macOS 14.1.1
Memory: 24 GiB
Architecture: aarch64

**Comment by arusahni** [comment]
(I'm putting this ask here because it feels related, please let me know if it is better suited as a distinct issue)

I'd also love the opposite option here: to have the option to not indent soft-wrapped lines, and instead start them from column 0 (vim-style). For long lines that are heavily indented, the same-indentation wrapping can make it hard to read.

e.g., zed

![image](https://github.com/user-attachments/assets/6b2f78ab-916f-4799-8111-8b98f1937c7c)

vs neovim

![image](https://github.com/user-attachments/assets/0bec0b44-0e78-487c-9dd1-4fb2c8b6178b)

**Comment by axsuul** [comment]
Yep, wrapping on the same indentation makes it very hard to determine if it's wrapped

**Comment by gdvine** [comment]
Thank you for creating this Issue, +1

**Comment by TicTocCroc** [comment]
Adding +1 on this feature. The addition of this would let me make Zed my daily driver.

**Comment by dumski** [comment]
I think would be even better to allow configuring boundaries for soft wrap, e.g. word boundaries (easier to implement) or even syntax-based boundaries (scoped_type_identifier, arguments, etc.)

**Comment by mfakhrys** [comment]
Any update on this? One year has passed since initial question..

**Comment by 20manas** [comment]
When wrapped relative lines are enabled, then it becomes even more to determine whether a line is soft-wrapped or not. So either some indentation or alternatively **some indicator beside the line number** should be useful to have.

**Comment by injust** [comment]
> When wrapped relative lines are enabled

How do relative line numbers make a difference here?

You still get this gap in the line numbers when a line is wrapped, whether relative line numbers are enabled or not:

<img width="96" height="170" alt="Image" src="https://github.com/user-attachments/assets/3d3160a5-92ac-425c-8c5f-3989fe1e1702" />

**Comment by 20manas** [comment]
For relative line numbers, we have three options: Disabled, Enabled, Wrapped. When we select the Wrapped option, the wrapped lines look something like this:

<img width="333" height="129" alt="Image" src="https://github.com/user-attachments/assets/c1975f04-3cb0-439a-a896-947d469c6cdf" />

This can create confusion about which line is wrapped and which is another separate line. While having wrapped relative lines is useful for faster editing, it is also useful to know when the text belongs to the same line and when it belongs to another. So IMO, it should be useful to have some kind of indicator beside the line number which tells us if a line is wrapped.

**Comment by mdaeron** [comment]
Any updates on this feature request? This would be quite handy in other contexts such as LaTeX or Typst source code.

**Comment by tidux** [comment]
The ability to run these commands on markdown files is literally the last thing keeping me from using Vim over Zed for writing:

```
  setlocal breakindentopt=list:-1
  setlocal breakindent
```

**Comment by arusahni** [comment]
I took a stab at this - it's up in #63468 . Zed's wrapping abstraction made it more straightforward than I thought.
